adapters/tappx/tappx.go

```diff
@@ -51,14 +51,58 @@ func (a *TappxAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapt
 			Message: "Error parsing bidderExt object",
 		}}
 	}
-
 	var tappxExt openrtb_ext.ExtImpTappx
 	if err := json.Unmarshal(bidderExt.Bidder, &tappxExt); err != nil {
 		return nil, []error{&errortypes.BadInput{
 			Message: "Error parsing tappxExt parameters",
 		}}
 	}
 
+	//In this section we create all EXT parameters
+	type VideoExt struct {
+		Rewarded int `json:"rewarded,omitempty"`
+	}
+	if tappxExt.Rewarded == 1 {
+		request.Imp[0].Video.Ext = json.RawMessage(`{"rewarded": 1}`)
+	}
+
+	type Bidder struct {
+		Tappxkey string   `json:"tappxkey"`
+		Mktag    string   `json:"mktag,omitempty"`
+		Bcid     []string `json:"bcid,omitempty"`
+		Bcrid    []string `json:"bcrid,omitempty"`
+	}
+	type Ext struct {
+		Bidder `json:"bidder"`
+	}
+	ext := Ext{
+		Bidder: Bidder{
+			Tappxkey: tappxExt.TappxKey,
+		},
+	}
+	if tappxExt.Mktag != "" {
+		ext.Bidder.Mktag = tappxExt.Mktag
+	}
+	if tappxExt.Bcid != nil {
+		ext.Bidder.Bcid = tappxExt.Bcid
+	}
+	if tappxExt.Bcrid != nil {
+		ext.Bidder.Bcrid = tappxExt.Bcrid
+	}
+
+	request.Ext = nil
+
+	jsonext, err := json.Marshal(ext)
+	if err == nil {
+		request.Ext = jsonext
+	}
```

Comment:
What if there is an error during marshaling? Would that still generate a valid Tappx request?

Reply:
Now yes, but in the future not.
Also, it is a good point to prevent now to the future.
Thank you!!!

We repair this in the 06a898d commit.